entity/c/c_secure_comm.h

```diff
@@ -0,0 +1,137 @@
+// @param purpose_length length of purpose
+// @param ret_length length of return buffer
+// @return concated total buffer
+unsigned char *auth_hello_reply_message(unsigned char *entity_nonce, unsigned char *auth_nonce, unsigned char num_key, unsigned char *sender, unsigned int sender_length, unsigned char *purpose, unsigned int purpose_length, unsigned int *ret_length);
+
+// Encrypt the message and sign the encrypted message.
+// @param buf input buffer
+// @param buf_len length of buf
+// @param path_pub public key path
+// @param path_priv private key path
+// @param message message with encrypted message and signature
+// @param message_length length of message
+unsigned char *encrypt_and_sign(unsigned char *buf, unsigned int buf_len, const char *path_pub, const char *path_priv, unsigned int *message_length);
+
+// Separate the message received from Auth and
+// store the distribution key in the distribution key struct
+// Must free distribution_key.mac_key, distribution_key.cipher_key
+// @param parsed_distribution_key distribution key struct to save information
+// @param buf input buffer with distribution key
+// @param buf_length length of buf
+void parse_distribution_key(distribution_key_t *parsed_distribution_key, unsigned char *buf, unsigned int buf_length);
+
+// Used in parse_session_key_response() for index.
+// @param buf input buffer with crypto spec
+// @param buf_length length of buf
+// @param offset buffer index
+// @param return_to_length length of return buffer
+// @return buffer with crypto spec
+unsigned char *parse_string_param(unsigned char *buf, unsigned int buf_length, int offset, unsigned int *return_to_length);
+
+// Store the session key in the session key struct
+// Must free when session_key expired or usage finished.
+// @param ret session key struct to save key info
+// @param buf input buffer with session key
+// @param buf_length length of buf
+// @return index number for another session key
+unsigned int parse_session_key(session_key_t *ret, unsigned char *buf, unsigned int buf_length);
+
+// Separate the session key, nonce, and crypto spec from the message.
+// @param buf input buffer with session key, nonce, and crypto spec
+// @param buf_length length of buf
+// @param reply_nonce nonce to compare with
+// @param session_key_list session key list struct
+void parse_session_key_response(unsigned char *buf, unsigned int buf_length, unsigned char *reply_nonce, session_key_t *session_key_list);
+
+// Generate the nonce to send to entity server,
+// encrypt the message with session key, and
+// make the total message including the session key id and encrypted message.
+// @param s_key session key struct to encrypt the message
+// @param entity_nonce nonce to protect the reply attack
+// @param ret_length length of return buffer
+// @return total buffer with session key id and encrypted message
+unsigned char *parse_handshake_1(session_key_t *s_key, unsigned char *entity_nonce, unsigned int *ret_length);
```

Comment:
The function's name seems incorrect based on the description in the comments. Can you update the function's name to match the description?